### DISCUSSION TITLE
Question: How to disable auto rearranging JSON data by alphabetical order?

### DISCUSSION THREAD

**Discussion by H1Mohamed** [pocketbase/pocketbase#5403] [Q&A]
I have some data in a json object, and I need them to stay always ordered, how can I stop it the database from rearranging the data by it self?
and if it doesn't exist, can we have it as an option in a future update?

**Comment by ganigeorgiev** [comment]
You can't, maps are unordered and Go doesn't guarantee to preserve the insertion order during iterations or serialization (_for the latter the default `encoding/json` will apply a deterministic alphabetical order_). See also https://github.com/pocketbase/pocketbase/discussions/4922#discussioncomment-9433526.
